RepositoryPattern-Starter/app/src/main/java/com/example/android/devbyteviewer/viewmodels/DevByteViewModel.kt

```diff
@@ -95,25 +91,26 @@ class DevByteViewModel(application: Application) : AndroidViewModel(application)
      */
     init {
         // TODO: Replace with a call to the refreshDataFromRepository9) method
-        refreshDataFromNetwork()
+        refreshDataFromRepository()
     }
 
     /**
      * Refresh data from the repository. Use a coroutine launch to run in a
      * background thread.
      */
     // TODO: Replace with the refreshDataFromRepository() method
-    private fun refreshDataFromNetwork() = viewModelScope.launch {
-        try {
-            val playlist = DevByteNetwork.devbytes.getPlaylist()
-            _playlist.postValue(playlist.asDomainModel())
-
-            _eventNetworkError.value = false
-            _isNetworkErrorShown.value = false
-
-        } catch (networkError: IOException) {
-            // Show a Toast error message and hide the progress bar.
-            _eventNetworkError.value = true
+    private fun refreshDataFromRepository() {
+        viewModelScope.launch {
+            try {
+                videosRepository.refreshVideos()
+                _eventNetworkError.value = false
+                _isNetworkErrorShown.value = false
+
+            } catch (networkError: IOException) {
+                // Show a Toast error message and hide the progress bar.
+                if(playlist.value.isNullOrEmpty())
+                    _eventNetworkError.value = true
+            }
```

Comment:
You replaced
```
private fun funcName() = viewModelScope.launch
...
```
by

```
private fun funcName() {
   viewModelScope.launch {
...
```

What that intentional? Could you please explain that?

Reply:
I just followed the tutorial, what I understood from this part was, the idea of the repository is to create an abstraction layer to obtain the data, that is, whoever is going to use this data doesn't need to know where it came from

Reply:
now talking about the difference in the return of the functions, in the first case the declaration of a function that returns the reference of a coroutine happens and in the second it executes the coroutine as soon as it is called

Reply:
this asynchronism in kotlin is still a bit confusing for me

Reply:
I think I have figured that out. The following declarations are equivalent:
```
    fun test1() = scope.launch { }
    fun test2(): Job {
        return scope.launch { }
    }
```

As `refreshDataFromNetwork()` returned a `Job` that was used for nothing, replacing that by `refreshDataFromRepository()` that returns `Unit` is ok.